src/App/components/Login.jsx

```diff
@@ -8,47 +8,57 @@ export default function Login({
   validationFunction,
 }) {
   // State variables for login screen
-  const [participantId, setParticipant] = React.useState(initialParticipantID);
-  const [studyId, setStudy] = React.useState(initialStudyID);
+  const [participantID, setParticipantID] = React.useState(initialParticipantID);
+  const [studyID, setStudyID] = React.useState(initialStudyID);
   const [isError, setIsError] = React.useState(false);
 
+  // Update local participantID if it changes upstream
+  useEffect(() => {
+    setParticipantID(initialParticipantID);
+  }, [initialParticipantID]);
+
+  // Update local studyID if it changes upstream
+  useEffect(() => {
+    setStudyID(initialStudyID);
+  }, [initialStudyID]);
+
```

Comment:
Why would it change upstream?

Reply:
I query parameters weren't auto-filling without these `useEffect`. I think it's because the state in `App.jsx` wasn't being set on the first render. I don't imagine folks will change the search parameters but if they do it will change the login fields which I think is nice